### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by derisen**
* Updated to msal.js 1.2.1 and added SRI hash.
* Source code refactored. Resource access decoupled from login process.
* Added new feature for reading user mails through MS Graph API.
* UI redesigned with Bootstrap 4 framework.
* Added AppCreationScripts for Powershell users.
* Added tokenization for for sovereign clouds and graph endpoints.
* Updated apps.json

### CODE REVIEW

File: AppCreationScripts/apps.json [hunk 1]
```diff
@@ -36,11 +36,12 @@
...
-            "clientId": "Enter_the_Application_Id_Here",
-            "authority": "Enter_the_Cloud_Instance_Here/Enter_the_Tenant_Info_Here",
-            "redirectUri": "Enter_the_Redirect_Uri_Here",
-            "graphMeEndpoint": "Enter_the_Graph_Endpoint_Here/v1.0/me",
-            "graphMailEndpoint": "Enter_the_Graph_Endpoint_Here/v1.0/me/messages"
+              "appId": "Enter_the_Application_Id_Here",
+              "tenantId": "Enter_the_Tenant_Info_Here",
+              "authorityEndpointHost": "Enter_the_Cloud_Instance_Id_Here",
+              "redirectUri": "Enter_the_Redirect_Uri_Here",
+              "msgraphEndpointHost": "Enter_the_Graph_Endpoint_Here/v1.0/me",
```
Comment (sameerag) [inline AppCreationScripts/apps.json]: Can we have two separate objects for configuration for token acquisition (msal js) and token usage(graph endpoint here)? msal js is designed to take a config object for initialization and appending resource end points to this confuses users. This is the same problem we are trying to solve for B2C samples too.

**Comment by sameerag** [comment]
Also, some code in auth.js is still referring to resource access, eg: ReadEmail should be ideally in graph.js and the code needs to be readjusted on what the interface between auth.js and graph.js will look like. This is *exactly* what we are trying to do in dev samples for `msal js` going forward FYI, so appreciate this approach choice for Azure AD Samples.

File: AppCreationScripts/apps.json (hunk 1 above)
Comment (derisen) [reply AppCreationScripts/apps.json]: I see -good idea.

Comment (derisen) [reply AppCreationScripts/apps.json]: applied the changes.

File: JavaScriptSPA/authConfig.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+ 
+  // Config object to be passed to Msal on creation
+  const msalConfig = {
+    auth: {
+      clientId: "Enter_the_Application_Id_Here",
+      authority: "Enter_the_Cloud_Instance_Id_Here/Enter_the_Tenant_Info_Here",
```
Comment (archieag) [inline JavaScriptSPA/authConfig.js]: The Azure portal replaces the endpoint host URL with protocol+domain+trailing slash. Update this to "Enter_the_Cloud_Instance_Id_HereEnter_the_Tenant_Info_Here"

File: JavaScriptSPA/graphConfig.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+  // Add here the endpoints for MS Graph API services you would like to use.
+  const graphConfig = {
```
Comment (archieag) [inline JavaScriptSPA/graphConfig.js:2]: The Azure portal replaces the endpoint host URL with protocol+domain+trailing slash. Update this to so the URL won't break.

**Comment by DarylThayil** [comment]
@pkanher617  since you are driving this change for portal updates, I will depend on you to be the final merge here. Please keep @archieag  timelines in mind

File: AppCreationScripts/apps.json (hunk 1 above)
Comment (archieag) [inline AppCreationScripts/apps.json]: "msgraphEndpointHost":"Enter_the_Graph_Endpoint_Here" is enough. The entire string is replaced with host url(protocol+domain+trailing slash and you would want to keep "v1.0/me".

File: AppCreationScripts/apps.json [hunk 4]
```diff
@@ -36,11 +36,12 @@
...
+              "msgraphEndpointHost:": "Enter_the_Graph_Endpoint_Here/v1.0/me/messages"
```
Comment (archieag) [inline AppCreationScripts/apps.json]: Repetitive. Please remove.

**Comment by sameerag** [comment]
One general q: Why powershell?

File: AppCreationScripts/apps.json (hunk 1 above)
Comment (derisen) [reply AppCreationScripts/apps.json]: sorry, that was the outdated file. recent pushes fixed this.

File: AppCreationScripts/apps.json (hunk 4 above)
Comment (derisen) [reply AppCreationScripts/apps.json]: same, outdated file. recent pushes fixed this.

**Comment by derisen** [comment]
> One general q: Why powershell?

ps scripts are for stuff like automatically registering the app on AAD portal, cleaning a previous registration etc. -they help during live demos.

File: JavaScriptSPA/authConfig.js (hunk 2 above)
Comment (mstada) [reply JavaScriptSPA/authConfig.js]: `Enter_the_Cloud_Instance_Id_Here` is not a supported token. The replacement token for this is `Enter_the_Authority_Endpoint_Host_Here`, so the whole authority is `"Enter_the_Authority_Endpoint_Host_HereEnter_the_Tenant_Info_Here"`.

File: AppCreationScripts/apps.json [hunk 5]
```diff
@@ -1,47 +1,55 @@
...
-       ]
-    }
-  ]
-}
+          },
+          {
+            "settingFile": "/JavaScriptSPA/graphConfig.js", 
+              "replaceTokens": 
+                {
+                "msgraphEndpointHost": "Enter_the_Graph_Endpoint_Here"
```
Comment (mstada) [inline AppCreationScripts/apps.json]: This token was changed to `Enter_the_MS_Graph_Endpoint_Host_Here`.

Comment (mstada) [reply AppCreationScripts/apps.json]: Please disregard.

**Review by mstada — APPROVED** [review]

File: JavaScriptSPA/authConfig.js (hunk 2 above)
Comment (mstada) [reply JavaScriptSPA/authConfig.js]: Please disregard

**Review by pkanher617 — APPROVED** [review]

**Pull request merged**